### ISSUE TITLE
stopped working again

### ISSUE THREAD

**Issue by ceeprus** [EvHaus/youtube-hide-watched#235]
doesnt hide watched videos anymore

only works on the sidebar when you watching videos

**Comment by EvHaus** [comment]
Still working well for me. Can you be more specific about what doesn't work? Which page are you on?

**Comment by ceeprus** [comment]
Subs page and homepage, it only works for video sidebars and inside playlists

**Comment by netjeff** [comment]
Still working fine for me on the subscriptions page at https://www.youtube.com/feed/subscriptions  
- Firefox 128.0.3 (64-bit) on Windows 11
- Tampermonkey 5.1.1
- YTHWV 6.3
- Plus several other extensions (ublock origin, noscript, cookie autodelete, etc)

**Comment by ceeprus** [comment]
Switched into a different youtube account and disabled seeing already watched vids fixed the homepage for me.
But subscriptions page still doesnt hide any vids

**Comment by ceeprus** [comment]
is there anyone can replicate what i have here? i really dont know what else to do.

**Comment by MagicfTail** [comment]
I'm having the same issue as @ceeprus, with videos not being greyed out/hidden on the main page and subscribers page.
Having very quickly looked into it, the issue appears to be that none of the following selectors are valid on line 275

```
watchedItem = (
	// Grid item
	item.closest('.ytd-grid-renderer') ||
	item.closest('.ytd-item-section-renderer') ||
	item.closest('.ytd-rich-grid-row') ||
	// List item
	item.closest('#grid-container')
);
```

My quick and dirty fix was adding the case `item.closest('.ytd-rich-grid-renderer')` as that appears to be the container related to the video "card" for my version of YouTube, but I have not throughly tested this fix. Let me know if there's anything I can provide that might help.

- Chrome 127.0.6533.72 (Official Build) (64-bit)
- Violentmonkey 2.20.0
- YTHWV 6.3
- Denmark
- Tested in incognito with no other extensions enabled

**Comment by ceeprus** [comment]
Yes! this fixes my problem

**Comment by MagicfTail** [comment]
I don't think the issue should be closed as the problem hasn't actually been solved. It would be nice if @EvHaus could have a look at it as well

**Comment by xavivars** [comment]
Same problem, and adding `item.closest('.ytd-rich-grid-renderer')` to the script also solved it.  Thanks @MagicfTail !

**Comment by fr05ty-mc** [comment]
for me today it has stopped showing the controls on any youtube pages (chrome and edge, firefox it still appears) but it is still enabled as the individual channel pages are hiding watched shows, also instead of leaving a blank tile it has started compressing them up next to each other not sure if that's a side effect or intended behavior.

i haven't added the above fix in to the script yet as I'm not real sure where to ad it

**Comment by ptc96** [comment]
> for me today it has stopped showing the controls on any youtube pages (chrome and edge, firefox it still appears) but it is still enabled as the individual channel pages are hiding watched shows, also instead of leaving a blank tile it has started compressing them up next to each other not sure if that's a side effect or intended behavior.

I've had the same issue since yesterday, the controls are just not present, but the before selected function still applies (after i added the above fix), i tried re-installing everything and cleaning all data from the browser but it doesn't work. It did work however in incognito (chrome) but when i tried to log into my account (still in incognito) it disappears...
Since i re-installed it, it is now set to normal and cannot change to hide o dimm.

**Comment by EvHaus** [comment]
Published v6.4 that should fix this.

**Comment by fr05ty-mc** [comment]
> > for me today it has stopped showing the controls on any youtube pages (chrome and edge, firefox it still appears) but it is still enabled as the individual channel pages are hiding watched shows, also instead of leaving a blank tile it has started compressing them up next to each other not sure if that's a side effect or intended behavior.
> 
> I've had the same issue since yesterday, the controls are just not present, but the before selected function still applies (after i added the above fix), i tried re-installing everything and cleaning all data from the browser but it doesn't work. It did work however in incognito (chrome) but when i tried to log into my account (still in incognito) it disappears... Since i re-installed it, it is now set to normal and cannot change to hide or dimm.

same here with the incognito tab (or a normal tab) as soon as i log on the controls vanish, log out and they come back.

other than that 6.4 is working with hiding the videos on the sub page again

**Comment by ptc96** [comment]
> same here with the incognito tab (or a normal tab) as soon as i log on the controls vanish, log out and they come back.

Updated to 6.4 cannot see the controls... I tried with another chrome profile and account and it works... This is beyond my comprehension at this poit 🙃 Could it be just us two? lol

**Comment by EvHaus** [comment]
It's very likely YouTube is gradually rolling out updates to their UI. Please give it a few more days. Once the changes are widely released, I'll fix it.

**Comment by ptc96** [comment]
@EvHaus Thank you, I really hope I didn't bother you too much 🙏🏻 it's the best subscription hider extension there is!

**Comment by fr05ty-mc** [comment]
> It's very likely YouTube is gradually rolling out updates to their UI. Please give it a few more days. Once the changes are widely released, I'll fix it.

Yep will do, I've tried chrome, edge and Firefox and FF is the only browser that shows the controls whilst logged in to my account, so I'll just use FF for now

Microsoft Edge
Version 127.0.2651.86 (Official build) (64-bit)

Chrome
Version 127.0.6533.89 (Official Build) (64-bit)

Firefox
128.0.3 (64-bit)

**Comment by EvHaus** [comment]
Missing buttons should be fixed in v6.5

**Comment by ptc96** [comment]
It works! You're the best! 🙏🏻
